[WIP] Get started docs

stevhliu opened the pull request:
# What does this PR do?

This PR contains the first draft of the Get Started docs as discussed for the new Transformers IA. Some additional thoughts:

- I think adding better signposts on the landing page will help users quickly get to the content they are looking for. For inspiration, maybe we can have cards like [these](https://stripe.com/docs/payments?payments=popular) with links to the pages (cc @mishig25). From the analytics, it looks like people view fine-tuning a pretrained model, fine-tuning for common downstream tasks, installation, and the BERT docs a lot. The first two pages definitely look valuable for getting users up and running with their tasks, but I would love to hear what other pages you think we can add to help guide users down the right path.

- On the Quick tour page, it'd be neat if we can embed a widget there (cc @mishig25) for sentiment analysis so users can play with it directly.

Commit:
first draft of get started docs

sgugger commented on docs/source/installation.md:
```diff
@@ -16,183 +16,182 @@ limitations under the License.
+</Tip>
 
-Now, let's get to the real benefit of this installation approach. Say, you saw some new feature has been just committed into `master`. If you have already performed all the steps above, to update your transformers to include all the latest commits, all you need to do is to `cd` into that cloned repository folder and update the clone to the latest version:
+Now you can easily update your clone to the latest version of `transformers` with the following commands:
```
```suggestion
Now you can easily update your clone to the latest version of 🤗 Transformers with the following commands:
```

sgugger commented on docs/source/installation.md:
````diff
@@ -16,183 +16,182 @@ limitations under the License.
 ```
 
-There is nothing else to do. Your python environment will find the bleeding edge version of `transformers` on the next run.
+Your Python environment will find the `master` version of `transformers` on the next run.
````
```suggestion
Your Python environment will find the `master` version of 🤗 Transformers on the next run.
```

sgugger commented on docs/source/installation.md:
````diff
@@ -16,183 +16,182 @@ limitations under the License.
 conda install -c huggingface transformers
 ```
 
-Follow the installation pages of TensorFlow, PyTorch or Flax to see how to install them with conda.
````
I would keep a mention of that.

stevhliu replied:
Since we linked to the installation pages for the various libraries at the top, do you think it'd be safe to assume users will have already installed `torch` or `tf` with either conda or pip? This way, we don't have to point users to the installation pages again. Or we could also drop this in as a **Tip** as a reminder?

sgugger commented on docs/source/installation.md:
```diff
@@ -16,183 +16,182 @@ limitations under the License.
+Pretrained models are downloaded and locally cached at: `~/.cache/huggingface/transformers/`. This is the default directory given by the shell environment variable `TRANSFORMERS_CACHE`. You can change the shell environment variables shown below - in order of priority - to specify a different cache directory:
 
-### Offline mode
+1. Shell environment variable: `HF_HOME`.
+2. Shell environment variable: `XDG_CACHE_HOME` + `/huggingface/`.
+3. Shell environment variable (default): `TRANSFORMERS_CACHE`.
```
This is wrong, `TRANSFORMERS_CACHE` as priority, so this list should be:

1. Shell environment variable `TRANSFORMERS_CACHE`
2. Shell environment variable `HF_HOME` + `/transformers/`
3. Shell environment variable: `XDG_CACHE_HOME` + `/huggingface/transformers`
4. Default `~/.cache/huggingface/transformers/`

stevhliu replied:
> these models will automatically be downloaded in the folder given by the shell environment variable TRANSFORMERS_CACHE. The default value for it will be the Hugging Face cache home followed by /transformers/.

Ah ok! This part here from the current docs makes it sound like the default value of `TRANSFORMERS_CACHE` is `~/.cache/huggingface/transformers/`. So are **1** and **4** the same then (sorry if I'm missing something super obvious here 😅 )?

sgugger commented on docs/source/installation.md:
```diff
@@ -16,183 +16,182 @@ limitations under the License.
+</Tip>
 
-Setting environment variable `TRANSFORMERS_OFFLINE=1` will tell 🤗 Transformers to use local files only and will not try to look things up.
+# Offline mode
```
Needs to be an h2 at least.

sgugger commented on docs/source/installation.md:
```diff
@@ -16,183 +16,182 @@ limitations under the License.
 
-See the reference for these methods in the huggingface_hub
-[documentation](https://github.com/huggingface/huggingface_hub/tree/main/src/huggingface_hub).
+    ![download-icon](/imgs/download.png)
```
Images should not be put in this folder anymore. See [here](https://github.com/huggingface/transformers/tree/master/docs#adding-an-image)

sgugger commented on docs/source/installation.md:
````diff
@@ -16,183 +16,182 @@ limitations under the License.
 
-It contains a set of tools to convert PyTorch or TensorFlow 2.0 trained Transformer models (currently contains `GPT-2`,
-`DistilGPT-2`, `BERT`, and `DistilBERT`) to CoreML models that run on iOS devices.
+        ```bash
+        python -m pip install huggingface_hub
+        ```
````
Nit: we only need to indent this under the 1. no?

stevhliu replied:
Indenting it further makes it a part of the preceding list item (looks a bit better this way I think) whereas indenting under `1.` would not:

<img width="478" alt="Screen Shot 2021-12-20 at 5 00 45 PM" src="https://user-images.githubusercontent.com/59462357/146843984-b15f17bd-afad-4f69-8c65-581e4bdb0657.png">

sgugger commented on docs/source/installation.md:
````diff
@@ -16,183 +16,182 @@ limitations under the License.
+
+    2. Use the [`hf_hub_download`](https://huggingface.co/docs/hub/adding-a-library#download-files-from-the-hub) function to download a file to a specific path. For example, here is how you can download the `config.json` file from the [T0](https://huggingface.co/bigscience/T0_3B) model to your desired path:
+
+        ```python
+        >>> from huggingface_hub import hf_hub_download
+        >>> hf_hub_download(repo_id="bigscience/T0_3B", filename="config.json", cache_dir="./your/path")
+        ```
````
Nit: we only need to indent this under the 2. no?

sgugger commented on docs/source/_toctree.yml:
```diff
@@ -1,14 +1,10 @@
-    title: Quick tour
   - local: installation
     title: Installation
-  - local: philosophy
-    title: Philosophy
```
Why are we removing this one? I thought the plan was to move it only?

stevhliu replied:
Sorry, got a little ahead of myself! 😅 Adding it back in, and we can move it to the Concept Guides once that is ready.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
 </Tip>
 
-## Getting started on a task with a pipeline
+# Pipeline
```
```suggestion
## Pipeline
```

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
+For more than one sentence, pass a list of sentences to the [`pipeline`] which returns a list of dictionaries:
 
-```py
+```python
````
py is perfectly valid here, there is no reason to switch to python.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
-fairly neutral.
+```python
+>>> from transformers import pipeline
+>>> pipe = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h", device=0)
````
Two things:
- `device=0` has not been explained yet
- Let's give this a descriptive name and not `pipe`. `speech-recognizer` for instance.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
-You can see the second sentence has been classified as negative (it needs to be positive or negative) but its score is
-fairly neutral.
+```python
+>>> from transformers import pipeline
````
We generally style doc samples to have one empty line between import and code (like black would do).

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
-look at its [model page](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) to get more
-information about it. It uses the [DistilBERT architecture](model_doc/distilbert) and has been fine-tuned on a
-dataset called SST-2 for the sentiment analysis task.
+Load the dataset you want to iterate over:
```
The documentation to `load_dataset` or the quicktour of Datasets should be linked here.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
+```
 
-You can directly pass the name of the model to use to [`pipeline`]:
+Now iterate over the dataset with the pipeline. This should be just as fast as running a custom loop on a GPU.
````
GPU support has not been introduced at this stage, and a random user running this code probably won't have one.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
+# as we're not interested in the _target_ part of the dataset.
+>>> for out in tqdm.tqdm(pipe(KeyDataset(dataset, "file"))):
+...    print(out)
+...    # {"text": "HE HOPED THERE WOULD BE STEW FOR DINNER TURNIPS AND CARROTS AND BRUISED POTATOES AND FAT MUTTON PIECES TO BE LADLED OUT IN THICK PEPPERED FLOWER FAT AND SAUCE"}
+...    # {"text": ....}
+...    # ....
```
If those are the outputs there should be not `...` in front of them.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
-```py
->>> classifier = pipeline('sentiment-analysis', model="nlptown/bert-base-multilingual-uncased-sentiment")
+```python
+# KeyDataset (only _pt_) will simply return the item in the dict returned by the dataset item
+# as we're not interested in the _target_ part of the dataset.
````
Let's not begin a code sample with a comment, we can put the same thing in prose before.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
+If you can't find a model for your use-case, you will need to fine-tune a pretrained model on your data. Take a look at our [fine-tuning tutorial](./training) to learn how. Finally, after you've fine-tuned your pretrained model, please consider sharing it (see tutorial [here](./model_sharing)) with the community on the Model Hub to democratize NLP for everyone! 🤗
 
-Let's now see what happens beneath the hood when using those pipelines.
+# AutoClasses
```
Header level 2 at least.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
+
+Let's return to our example and see how we can use the `AutoClasses` to replicate the results of the [`pipeline`].
+
+## Using the AutoTokenizer
```
Should be one more than before, so 3 here.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -14,186 +14,186 @@ specific language governing permissions and limitations under the License.
-```py
->>> inputs = tokenizer("We are very happy to show you the 🤗 Transformers library.")
+```python
+>>> input = tokenizer("We are very happy to show you the 🤗 Transformers library.")
````
`input` is a reserved word in Python so we should avoid using it. How about `encoding`?

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
-===PT-TF-SPLIT===
->>> tf_outputs = tf_model(tf_batch)
-```
+## Using the AutoModel
````
One more level

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
 ```
 
-We can see we get the numbers from before:
+The model outputs the model's final activations in the `logits` attribute. Apply the softmax activation to the `logits` to retrieve the predictions:
````
```suggestion
The model outputs the model's final activations in the `logits` attribute. Apply the softmax function to the `logits` to retrieve the probabilities:
```
Softmax is not an activation, and it returns probabilities. Predictions would be the classes with the maximum probabilities/logits.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
-Models are standard [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) or [tf.keras.Model](https://www.tensorflow.org/api_docs/python/tf/keras/Model) so you can use them in your usual training loop. 🤗 Transformers also provides a [`Trainer`] class to help with your training in PyTorch (taking care of things such as distributed training, mixed precision, etc.) whereas you c ...
-
 <Tip>
+All 🤗 Transformers models (PyTorch or TensorFlow) return the model activations *before* the final activation
+function (like softmax) because the final activation function is often fused with the loss.
```
```suggestion
All 🤗 Transformers models (PyTorch or TensorFlow) outputs the tensors *before* the final activation
function (like softmax) because the final activation function is often fused with the loss.
```

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
-Pytorch model outputs are special dataclasses so that you can get autocompletion for their attributes in an IDE.
-They also behave like a tuple or a dictionary (e.g., you can index with an integer, a slice or a string) in which
-case the attributes not set (that have `None` values) are ignored.
+Models are standard [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) or [tf.keras.Model](https://www.tensorflow.org/api_docs/python/tf/keras/Model) so you can use them in your usual training loop. However, to make things easier, 🤗 Transformers provides a [`Trainer`] class for PyTorch that adds functionality for distributed training, mixed precision, and more. For TensorF ...
```
```suggestion
Models are standard [`torch.nn.Module`](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) or [`tf.keras.Model`](https://www.tensorflow.org/api_docs/python/tf/keras/Model) so you can use them in your usual training loop. However, to make things easier, 🤗 Transformers provides a [`Trainer`] class for PyTorch that adds functionality for distributed training, mixed precision, and more. For TensorFlow, you can use the `fit()` method from [Keras](https://keras.io/). Refer to the [training tutorial](./training) for more details.
```

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
 </Tip>
 
-Once your model is fine-tuned, you can save it with its tokenizer in the following way:
+## Save your model
```
Probably need one level more.

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
-Once your model is fine-tuned, you can save it with its tokenizer in the following way:
+## Save your model
+
+Once your model is fine-tuned, you can save it with its tokenizer using `save_pretrained`:
```
Link to `PreTrainedModel.save_pretrained` here.
```suggestion
Once your model is fine-tuned, you can save it with its tokenizer using [`~PreTrainedModel.save_pretrained`]:
```

sgugger commented on docs/source/quicktour.mdx:
```diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
-You can then load this model back using the [`AutoModel.from_pretrained`] method by passing the
-directory name instead of the model name. One cool feature of 🤗 Transformers is that you can easily switch between
-PyTorch and TensorFlow: any model saved as before can be loaded back either in PyTorch or TensorFlow.
+When you are ready to use the model again, reload it with `from_pretrained`:
```
Same here, link to the doc.

sgugger commented on docs/source/quicktour.mdx:
````diff
@@ -211,220 +211,93 @@ and get tensors back. You can specify all of that to the tokenizer:
->>> all_attentions = tf_outputs.attentions
-```
-
-### Accessing the code
````
I don't think we should remove this from here if we are not adding it somewhere else.

stevhliu replied:
For sure, we will add this section in the guides as an alternative way to load a model.

sgugger reviewed (commented):
Thanks for starting your work on this!
I have left a few comments. In particular I'm not sure we should start with the top-level guides and remove content, even if temporarily.

stevhliu commented on docs/source/index.mdx:
```diff
@@ -12,81 +12,30 @@ specific language governing permissions and limitations under the License.
+- Get Started will introduce you to 🤗 Transformers and showcase it's ease-of-use and high performance. You'll install our library, use the `pipeline` API to quickly use a pretrained model for inference, and then preprocess and fine-tune your own model.
+- Tutorials are a good starting point if you've never used 🤗 Transformers before. Learn from the ground up how to preprocess text, image or audio data, initialize a pretrained model with the `AutoClass` API, fine-tune a model with our `Trainer` or deep learning library of your choice, enable distributed training with [Accelerate](https://huggingface.co/docs/accelerate/), and share a model to the  ...
+- Guides will show you how to apply your skills to solve real-world problems such as fine-tuning models for some of the most common downstream tasks, using multi-lingual models, and optimizing performance for larger models. If you are interested in contributing to our community and participating in open-source science, we will also show you how to add a model or pipeline to 🤗 Transformers.
+- Read the Concept guides to gain a deeper understanding about the 🤗 Transformers design philosophy, different kinds of Transformer models, types of tasks 🤗 Transformers can be applied to, and strategies for training increasingly large models.
```
Oops, got a little too ahead of myself here! 😅 Completely agree with you here, I'll revisit `index.mdx` last when all the content is there.

stevhliu commented on docs/source/_toctree.yml:
```diff
@@ -1,14 +1,10 @@
     title: Installation
-  - local: philosophy
-    title: Philosophy
-  - local: glossary
-    title: Glossary
```
Adding the Glossary back in, and saving it for when we actually have a Concept Guide (also made a note to check and update links to Glossary).

stevhliu commented on docs/source/_toctree.yml:
```diff
@@ -1,14 +1,10 @@
 - sections: 
   - local: index
     title: 🤗 Transformers
-  - local: quicktour
-    title: Quick tour
```
Hmm interesting, I was thinking it'd be better to show how to install so the user is all set up before doing the quicktour. Not a big deal either way though since we can just add `pip install transformers` to the quicktour. 👍

stevhliu commented on docs/source/installation.md:
```diff
@@ -16,183 +16,182 @@ limitations under the License.
 # Installation
 
-🤗 Transformers is tested on Python 3.6+, and PyTorch 1.1.0+ or TensorFlow 2.0+.
+Install 🤗 Transformers for whichever deep learning library you're working with, setup your cache, and configure 🤗 Transformers to run offline.
```
Changed to mention it's optional! I think I'll leave it in though so it's clear what the user can expect from this page 😊

Commit:
apply feedback from review

Commit:
add more edits

Commit:
rename md to mdx

stevhliu committed:
Merge branch 'master' into get-started-docs

stas00 committed:
[doc] install - add jax (#14912)

As `jax` cuda requires special instructions to be installed correctly add a link to jax installation instructions. 

Note: Flax install page only covers cpu jax installation info.

patrickvonplaten committed:
[WavLM] fix wavlm docs (#14910)

Sanster committed:
Fix Perceiver docs (#14917)

Narsil committed:
ChunkPipeline (batch_size enabled on `zero-cls` and `qa` pipelines. (#14225)

* Pipeline chunks.

* Batching for Chunking pipelines ?

* Batching for `question-answering` and `zero-shot-cls`.

* Fixing for FNet.

* Making ASR a chunk pipeline.

* Chunking ASR API.

* doc style.

* Fixing ASR test.

* Fixing QA eror (p_mask, padding is 1, not 0).

* Enable both vad and simple chunking.

* Max length for vad.

* remove inference mode, crashing on s2t.

* Revert ChunkPipeline for ASRpipeline.

Too many knobs for simple integration within the pipeline, better stick
to external convenience functions instead, more control to be had,
simpler pipeline and also easier to replace with other things later.

* Drop necessity for PT for these.

* Enabling generators.

* Add mic + cleanup.

* Typo.

* Typo2.

* Remove ASR work, it does not belong in this PR anymore.

* Update src/transformers/pipelines/pt_utils.py

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* Update src/transformers/pipelines/zero_shot_classification.py

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* Adding many comments.

* Doc quality.

* `hidden_states` handling.

* Adding doc.

* Bad rebase.

* Autofixing docs.

* Fixing CRITICAL bug in the new Zerocls pipeline.

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

stancld committed:
Add `ElectraForCausalLM` -> Enable Electra encoder-decoder model (#14729)

* Add ElectraForCausalLM and cover some basic tests & need to fix a few tests

* Fix bugs

* make style

* make fix-copies

* Update doc

* Change docstring to markdown format

* Remove redundant update_keys_to_ignore

Commit:
fix to issue #14833 in data_collator - consider no labels (#14930)

MihaiBalint committed:
Fix duplicate call to save_checkpoint when using deepspeed (#14946)

* Fix duplicate call to save_checkpoint when using deepspeed / stage3_gather_fp16_weights_on_model_save

* Revert "Fix duplicate call to save_checkpoint when using deepspeed / stage3_gather_fp16_weights_on_model_save"

This reverts commit 6a3dec0397723a8417351dc38fdebf14ab17756c.

* Delete correct duplicate invocation of deepspeed save_checkpoint

sgugger committed:
Doc styler v2 (#14950)

* New doc styler

* Fix issue with args at the start

* Code sample fixes

* Style code examples in MDX

* Fix more patterns

* Typo

* Typo

* More patterns

* Do without black for now

* Get more info in error

* Docstring style

* Re-enable check

* Quality

* Fix add_end_docstring decorator

* Fix docstring

sgugger committed:
Convert last rst file (#14952)

stas00 committed:
[doc] consistent True/False/None default format (#14951)

* [doc] consistent True/False/None default format

* Update src/transformers/models/xlnet/modeling_xlnet.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

stas00 committed:
[doc] :obj: hunt (#14954)

* redo sans examples

* style

sgugger committed:
Doc styler examples (#14953)

* Fix bad examples

* Add black formatting to style_doc

* Use first nonempty line

* Put it at the right place

* Don't add spaces to empty lines

* Better templates

* Deal with triple quotes in docstrings

* Result of style_doc

* Enable mdx treatment and fix code examples in MDXs

* Result of doc styler on doc source files

* Last fixes

* Break copy from

rebase

sgugger committed:
Style

stas00 committed:
[doc] :class: hunt (#14955)

* [doc] :class: hunt

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* fix the fix + style

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

patrickvonplaten committed:
Add Speech Seq2Seq Training script (#14792)

* start

* add gradient checkpointing and feature extractor freezing

* Apply suggestions from code review

* up

* up

* up

* correct

* up

* more changes

* up

* up

* up

* remove rst

patrickvonplaten committed:
[WavLM] give model for precision (#14958)

patrickvonplaten committed:
Update README.md (#14965)

patrickvonplaten committed:
[Tests] Speed up tokenizer tests (#14964)

* speed up canine and mluke

* speed up mbart and mbart50 toks

* upload files

patrickvonplaten committed:
[Wav2Vec2] Rename model's feature extractor to feature encoder (#14959)

* rename classes

* clean up more namings

* remove bogus file

* Apply suggestions from code review

* Apply suggestions from code review

* replace more names

* more regex replace

* make style

* correct

* correct more

* make style

* finish

* correct more in wav2vec2

* make style

* improve freeze_extractor

* add aliases

* add tf aliases

jaketae committed:
refactor: replace `assert` with `ValueError` (#14970)

stas00 committed:
remove absl workaround as it's no longer needed (#14909)

the absl workaround hasn't been needed since 2019-04 https://github.com/abseil/abseil-py/issues/99 so it should be safe to remove it.

Narsil committed:
Fixing a pathological case for slow tokenizers (#14981)

* Fixing a pathological case for slow tokenizers

* Update src/transformers/tokenization_utils.py

patrickvonplaten committed:
[AutoProcessor] Correct AutoProcessor and automatically add processor… (#14881)

* [AutoProcessor] Correct AutoProcessor and automatically add processor class

* up

* up

* up

* up

* up

* up

* up

* up

* continue tomorrow

* up

* up

* up

* make processor class private

* fix loop

patrickvonplaten committed:
[Generate] correct encoder_outputs are passed without attention_mask (#14980)

* [Generate] correct encoder_outputs are passed without attention_mask

* Apply suggestions from code review

* up

Narsil committed:
Adding `num_return_sequences` support for text2text generation. (#14988)

* Adding `num_return_sequences` support for text2text generation.

Co-Authored-By: Enze <pu.miao@foxmail.com>

* Update tests/test_pipelines_text2text_generation.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update tests/test_pipelines_text2text_generation.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Enze <pu.miao@foxmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Narsil committed:
Enabling `tokenizers` upgrade. (#14941)

* Enabling `tokenizers` upgrade.

* Moved ugly comment.

* Tokenizers==0.11.1 needs an update to keep borrow checker

happy in highly contiguous calls.

* Support both 0.11.1 and 0.11.0

sgugger committed:
Allow training to resume even if RNG states are not properly loaded (#14994)

* Allow training to resume even if RNG states are not properly loaded

* Proper f-string

sgugger committed:
Map model_type and doc pages names (#14944)

* Map model_type and doc pages names

* Add script

* Fix typo

* Quality

* Manual check for Auto

Co-authored-by: Lysandre <lysandre.debut@reseau.eseo.fr>

Narsil committed:
Fixing t2t pipelines lists outputs. (#15008)

Backward compatibility broken in
https://github.com/huggingface/transformers/pull/14988

Narsil committed:
Improve truncation_side (#14947)

* Enabling `truncation_side` for Slow and Fast tokenizer.

Co-Authored-by: Niels Rogge <48327001+NielsRogge@users.noreply.github.com>

* Disable failing tests.

* Layout xlm.

* assert -> assertEqual.

Co-authored-by: Niels Rogge <48327001+NielsRogge@users.noreply.github.com>

anton-l committed:
Large audio chunking for the existing ASR pipeline (#14896)

* Naive ASR chunking

* Fixing batching for ASR.

Co-authored-by: Nicolas Patry <patry.nicolas@protonmail.com>

ydshieh committed:
fix missing import (#15016)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

patrickvonplaten committed:
[Tests] Correct Wav2Vec2 & WavLM tests (#15015)

* up

* up

* up

hyunwoongko committed:
Update parallelism.mdx (#15013)

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

* Update parallelism.mdx

flozi00 committed:
Fix Code block (#14983)

milyiyo committed:
Fix a little typo (#15002)

stancld committed:
Add Flax RoFormer (#15005)

* Add FlaxRoFormer

* Clean code + make quality

* Fix output pooling for FlaxRoFormerForMultipleChoiceModule

* Apply suggestions from code review

* add flax model to repos

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

Narsil committed:
Hotfix `chunk_length_s` instead of `_ms`. (#15029)

* Hotfix `chunk_length_s` instead of `_ms`.

* Adding fix of `pad_token` which should be last/previous token for CTC

proper decoding

* Fixing ChunkPipeline unwrapping.

* Adding a PackIterator specific test.

hyunwoongko committed:
[doc] Update parallelism.mdx (#15018)

* Update parallelism.mdx

* Update parallelism.mdx

LysandreJik commented on docs/source/installation.mdx:
```diff
@@ -16,183 +16,189 @@ limitations under the License.
-You should install 🤗 Transformers in a [virtual environment](https://docs.python.org/3/library/venv.html). If you're
-unfamiliar with Python virtual environments, check out the [user guide](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/). Create a virtual environment with the version of Python you're going
-to use and activate it.
+🤗 Transformers is tested on Python 3.6+, PyTorch 1.1.0+, TensorFlow 2.0+, and Flax. Follow the installation instructions below for the deep learning library you are using:
```
which version of Flax @patil-suraj ?

LysandreJik commented on docs/source/installation.mdx:
```diff
@@ -16,183 +16,189 @@ limitations under the License.
-  * shell environment variable ``HF_HOME``
-  * shell environment variable ``XDG_CACHE_HOME`` + ``/huggingface/``
-  * default: ``~/.cache/huggingface/``
+Pretrained models are downloaded and locally cached at: `~/.cache/huggingface/transformers/`. This is the default directory given by the shell environment variable `TRANSFORMERS_CACHE`. You can change the shell environment variables shown below - in order of priority - to specify a different cache directory:
```
This is only true for linux/mac. It would be nice to also put that for Windows - do you have access to a Windows machine?

stevhliu replied:
Good idea, and yes I have a Windows machine! Would the default directory on Windows be `C:\Users\username\.cache\huggingface\transformers`?

LysandreJik replied:
That's why I'm asking, I don't know :)

LysandreJik commented on docs/source/installation.mdx:
```diff
@@ -16,183 +16,189 @@ limitations under the License.
+
+### Fetch models and tokenizers to use offline
+
+Another option for using 🤗 Transformers offline is to download the files ahead of time, and then point to their local path when you need to use them offline. There are two ways to do this:
```
A third option is to use `from_pretrained` first, then `save_pretrained` and have the future `from_pretrained` calls point to the directory specified in the `save_pretrained`

stevhliu replied:
Awesome, I'll include this as well!

LysandreJik commented on docs/source/installation.mdx:
````diff
@@ -16,183 +16,189 @@ limitations under the License.
+>>> config = AutoConfig.from_pretrained("./your/path/bigscience_t0/config.json")
+```
 
-At some point in the future, you'll be able to seamlessly move from pretraining or fine-tuning models in PyTorch or
-TensorFlow 2.0 to productizing them in CoreML, or prototype a model or an app in CoreML then research its
-hyperparameters or architecture from PyTorch or TensorFlow 2.0. Super exciting!
````
cc @julien-c, the end of an era

LysandreJik reviewed (commented):
Excited for this!

stas00 committed:
[megatron convert] PYTHONPATH requirements (#14956)

* [megatron convert] PYTHONPATH requirements

* more info

ydshieh committed:
Fix doc example: mask_time_indices (numpy) has no attribute 'to' (#15033)

* fix doc example - AttributeError: 'numpy.ndarray' object has no attribute 'to'

* fix more

* Apply suggestions from code review

* Update src/transformers/models/unispeech/modeling_unispeech.py

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

Narsil committed:
Adding QoL for `batch_size` arg (like others enabled everywhere). (#15027)

* Adding QoL for `batch_size` arg (like others enabled everywhere).

* Typo.

patrickvonplaten committed:
[CLIP] Fix PT test (#15041)

patrickvonplaten committed:
[SpeechEncoderDecoder] Fix from pretrained (#15043)

patil-suraj committed:
[CLIP] Fix TF test (#15042)

Commit:
add third option for offline loading

LysandreJik commented:
Hey Steven, I believe there was an issue in your merge/rebase and GitHub is having a hard time understanding what happened. Could you close and reopen a new PR without touching at your branch so that we may see the actual differences? Thank you.
